### DISCUSSION TITLE
Autorefresh not enabled by default

### DISCUSSION THREAD

**Discussion by RafaelLinux** [openSUSE/opi#183] [Q&A]
For me "opi" has become the tool to install everything I need that is not in openSUSE repositories. I usually use it after every new openSUSE installation, I use the "`-nm`" parameters and I want all added repositories to have enabled "refresh".

However, with each new installation, I notice that the "refresh" appears as disabled in the repositories it adds.

In [your website](https://github.com/openSUSE/opi) it says that by default, the value `new_repo_auto_refresh=true`, because I have not assigned its value previously. However, the "refresh" is not being activated in the added repositories.

Am I doing something wrong? Is `new_repo_auto_refresh` an environment variable to assign in `bashrc`?

**Comment by aylen384** [comment]
The config file where you can change that values is `/etc/opi.cfg` - this is mine for example:

```ini
[opi]
backend = zypp
use_releasever_var = true
new_repo_auto_refresh = true
list_in_reverse = false
```

And when I invoke opi:
```
# opi xosview
Searching repos for: xosview
1. xosview
2. xosview-debuginfo
3. xosview-debugsource
Pick a number (0 to quit): 1
You have selected package name: xosview
1. Haupt-Repository (OSS) +                  | 1.24                      | x86_64
2. openSUSE:Factory -                        | 1.24                      | x86_64
3. X11:Utilities ?                           | 1.24                      | x86_64
Pick a number (0 to quit): 3
You have selected package: X11:Utilities ? | 1.24 | x86_64
Adding repo 'X11:Utilities'
[...]
```

I get the repo with autorefresh enabled (content of `/etc/zypp/repos.d/X11_Utilities.repo`):

```ini
[X11_Utilities]
name=X11:Utilities
baseurl=https://download.opensuse.org/repositories/X11:/Utilities/openSUSE_Tumbleweed/
enabled=1
type=rpm-md
gpgcheck=1
gpgkey=https://download.opensuse.org/repositories/X11:/Utilities/openSUSE_Tumbleweed/repodata/repomd.xml.key
autorefresh=1
```

To show, what value your variable has internally (note that python uses `True` and `False` internally, while in the config file `true` and `false` are used):
```
$ python3 -c "import opi; print(opi.config.get_key_from_config('new_repo_auto_refresh'))"
True
```

**Reply by RafaelLinux** [reply]
Thank you for explaining it to me. I didn't know there was such a configuration file. The curious thing is that when I run that command you put - without modifying anything in the configuration file - it returns ‘True’, that is to say, that in theory, it should activate ‘Autorefresh’ by default. Tomorrow I'll try it on another computer where I installed TW from 0 a couple of weeks ago, to see if it also returns ‘True’ (that's where I noticed the problem).

As a suggestion ... in the future it would be interesting to be able to change the ‘opi’ configuration with a parameter.
